packages/suite/src/components/suite/images/images.ts

```diff
@@ -109,63 +55,53 @@ const SVG_IMAGES = {
     EARLY_ACCESS_DISABLE: 'early-access-disable.svg',
 } as const;
 
-type pngImageKey = keyof typeof PNG_IMAGES;
-type svgImageKey = keyof typeof SVG_IMAGES;
-export type Props = React.ImgHTMLAttributes<Omit<HTMLImageElement, 'src'>> & {
-    image: pngImageKey | svgImageKey;
-    alt?: string; // why? Seems not to be part of HTMLImageElement :(
-};
-
-const buildSrcSet = <
-    BasePath extends string,
-    ImageObject extends typeof PNG_IMAGES | typeof SVG_IMAGES,
-    ImageKey extends keyof ImageObject,
->(
-    basePath: BasePath,
-    imageObject: ImageObject,
-    imageKey: ImageKey,
-) => {
-    const imageFile1x = imageObject[imageKey];
-    const hiRes = `${imageKey}_2x`;
-    const imageFile2x = hiRes in imageObject ? imageObject[hiRes as ImageKey] : undefined;
-    if (!imageFile2x) return undefined;
-
-    return `${resolveStaticPath(`${basePath}/${imageFile1x}`)} 1x, ${resolveStaticPath(
-        `${basePath}/${imageFile2x}`,
-    )} 2x`;
-};
+export type PngImage = keyof typeof PNG_IMAGES;
 
-const Image = styled.img`
-    /* should not overflow it's container */
-    max-width: 100%;
-
-    ${props =>
-        props.theme.IMAGE_FILTER &&
-        css`
-            filter: ${props.theme.IMAGE_FILTER};
-        `}
-`;
-
-const ImageComponent = ({ image, ...props }: Props) => {
-    if (image in PNG_IMAGES) {
```

Comment:
This seems like a perfect spot for typeguards. I'd do something like
```const isPNG = (image: ImageType): image is PngImage => image in PNG_IMAGES;```
and then just `if (isPNG(image)) {...` without `as` assertions.

Reply:
Yaaay, thanks, I finally used a typeguard 😅

Reply:
cbd767025bd9279bc2d96c7ad5381c64bd6c3767